Objects/genericaliasobject.c

```diff
@@ -626,6 +641,46 @@ static PyNumberMethods ga_as_number = {
         .nb_or = _Py_union_type_or, // Add __or__ function
 };
 
+static PyObject *
+ga_iternext(gaiterobject *gi) {
+    if (gi->obj == NULL) {
+        return NULL;
+    }
+    gaobject *alias = (gaobject *)gi->obj;
+    PyObject *starred_tuple = Py_GenericAlias(alias->origin, alias->args);
+    if (starred_tuple == NULL) {
+        return NULL;
+    }
+    ((gaobject *)starred_tuple)->starred = true;
+    Py_SETREF(gi->obj, NULL);
+    return starred_tuple;
+}
+
+static void
+ga_iter_dealloc(gaiterobject *gi) {
+    PyObject_GC_UnTrack(gi);
+    PyObject_GC_Del(gi);
```

Comment:
Consider the case of deallocating an iterator that hasn't been exhausted. We need another decref here.
```suggestion
    PyObject_GC_UnTrack(gi);
    Py_XDECREF(gi->obj);
    PyObject_GC_Del(gi);
```

Reply:
Ahh, good point! Fixed.